### PR TITLE
@pepopowitz => [Gene] Relax paging check which stops paging early

### PR DESCRIPTION
**Pull request by mzikherman**
Fixes https://artsyproduct.atlassian.net/secure/RapidBoard.jspa?rapidView=57&projectKey=DISCO&modal=detail&selectedIssue=DISCO-866

## Problem
We used to have 'issues' with infinite scroll where it would start going haywire and eventually start to make requests with really high page numbers. Ship bug to users, have them keep their browser tabs open, and you've just DDOS'ed yourself!

## History
So, we made this very conservative check where: if you ever got back _less_ than the page size, we would just stop paging. The assumption being, even if `hasNextPage` is incorrectly coming back true, once we hit an empty page we would stop.

## Solution
Relax this check somewhat to still protect us against DDOS and the infinite scroll component behaving badly, but allow for some occasional gaps in data and pages (as we seem to currently have).

## Caveats
This is just a band-aid around the fact that for some reason, on page 12 of the [Abstract Painting](https://www.artsy.net/gene/abstract-painting) gene, we wind up having an artwork returned _that already was returned_, and so Relay winds up not including that duplicate in the connection, and thus there's only 119 instead of 120 total artworks after the 12th page is fetched. I'm not sure why that's the case, but this feels like a reasonable change.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit d6e9de477e22bd5b36c51250327c0015ea7b6c6d

https://deploy-preview-2239--artsy-reaction.netlify.com

**Comment by mzikherman** [comment]
@pepopowitz that's a great idea, but I'd like to just merge this as-is since it fixes a high priority bug that's been reported for a little while already.

Your suggestion definitely makes sense, but it might be a bit more nuanced to get it perfectly right (which is why I would want to just merge as-is). For instance, there's a `componentDidMount` hook to start a throttled check: https://github.com/artsy/reaction/blob/c975f33b6158d7b31980876afaaa65660cb25368/src/Components/ArtworkGrid/ArtworkGrid.tsx#L61-L66

So, returning early from `maybeLoadMore` and not triggering `onLoadMore` here: https://github.com/artsy/reaction/blob/c975f33b6158d7b31980876afaaa65660cb25368/src/Components/ArtworkGrid/ArtworkGrid.tsx#L93 is definitely good! But...you still get the now redundant throttled check happening. It's probably better with a `clearInterval`  somewhere in there.

Basically I would adopt your suggestion as a refactor and possibly not lumped in this PR/bug-fix.

**Comment by artsyit** [comment]
:rocket: PR was released in v15.11.14 :rocket:
